lib/Extension/LanguageServerCompletion/Handler/CompletionHandler.php

```diff
@@ -96,21 +110,30 @@ public function completion(CompletionParams $params, CancellationToken $token):
 
                 $textEdits = $nameImporterResult->getTextEdits();
 
-                $items[] = CompletionItem::fromArray([
-                         'label' => $name,
-                         'kind' => PhpactorToLspCompletionType::fromPhpactorType($suggestion->type()),
-                         'detail' => $this->formatShortDescription($suggestion),
-                         'documentation' => new MarkupContent('markdown', $suggestion->documentation() ?? ''),
-                         'insertText' => $insertText,
-                         'sortText' => $this->sortText($suggestion),
-                         'textEdit' => $this->textEdit($suggestion, $insertText, $textDocument),
-                         'additionalTextEdits' => $textEdits,
-                         'insertTextFormat' => $insertTextFormat
-                     ]);
+                $item = CompletionItem::fromArray([
+                    'label' => $name,
+                    'kind' => PhpactorToLspCompletionType::fromPhpactorType($suggestion->type()),
+                    'insertText' => $insertText,
+                    'sortText' => $this->sortText($suggestion),
+                    'textEdit' => $this->textEdit($suggestion, $insertText, $textDocument),
+                    'additionalTextEdits' => $textEdits,
+                    'insertTextFormat' => $insertTextFormat,
+                    'data' => $index,
+                ]);
+
+                $this->resolve[$index] = function (CompletionItem $item) use ($suggestion): CompletionItem {
+                    $documentation = $suggestion->documentation();
+                    $item->documentation = $documentation ? new MarkupContent(MarkupKind::MARKDOWN, $documentation) : null;
+                    $item->detail = $this->formatShortDescription($suggestion);
```

Comment:
This removes the "eager" details, in Nvim I use this to show the FQN of the classes - but a better way would be to include the FQN (or at least an indicative abbreviation) as the `label`